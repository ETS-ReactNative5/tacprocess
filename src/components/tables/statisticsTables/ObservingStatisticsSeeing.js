import React from 'react'
import propTypes from 'prop-types'
import { LOW_PRECISION } from '../../../types'
import { getPercentage, rounded } from '../../../util'

const ObservingStatisticsSeeing = ({ seeingDistribution }) => {
  const { numberOfProposals, timeRequested } = seeingDistribution
  const totalReqTime = timeRequested.between0And1Dot5 + timeRequested.between1Dot5And2 +
    timeRequested.between2And2Dot5 + timeRequested.between2Dot5And3 + timeRequested.moreThan3

  return(
    <div>
      <table className='stat-table'>
        <thead>
          <tr>
            <th className='width-200'>Conditions</th>
            <th>Time requested (hrs)</th>
            <th>Number of Proposals</th>
            <th>Fraction of Total Request (%)</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>0 &le; Max Seeing &le; 1.5 </td>
            <td>{ rounded(timeRequested.between0And1Dot5) }</td>
            <td>{ numberOfProposals.between0And1Dot5 }</td>
            <td>{ rounded(getPercentage(timeRequested.between0And1Dot5, totalReqTime), LOW_PRECISION) }</td>
          </tr>
          <tr>
            <td>1.5 &lt; Max Seeing  &le; 2.0</td>
            <td>{ rounded(timeRequested.between1Dot5And2) }</td>
            <td>{ numberOfProposals.between1Dot5And2 }</td>
            <td>{ rounded(getPercentage(timeRequested.between1Dot5And2, totalReqTime), LOW_PRECISION) }</td>
          </tr>
          <tr>
            <td>2.0 &lt; Max Seeing &le; 2.5</td>
            <td>{ rounded(timeRequested.between2And2Dot5) }</td>
            <td>{ numberOfProposals.between2And2Dot5 }</td>
            <td>{ rounded(getPercentage(timeRequested.between2And2Dot5, totalReqTime), LOW_PRECISION) }</td>
          </tr>
          <tr>
            <td>2.5 &lt; Max Seeing &le; 3.0</td>
            <td>{ rounded(timeRequested.between2Dot5And3) }</td>
            <td>{ numberOfProposals.between2Dot5And3 }</td>
            <td>{ rounded(getPercentage(timeRequested.between2Dot5And3, totalReqTime), LOW_PRECISION) }</td>
          </tr>
          <tr>
            <td>Max Seeing &ge; 3.0</td>
            <td>{ rounded(timeRequested.moreThan3) }</td>
            <td>{ numberOfProposals.moreThan3 }</td>
            <td>{ rounded(getPercentage(timeRequested.moreThan3, totalReqTime), LOW_PRECISION) }</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

ObservingStatisticsSeeing.propTypes = {
  seeingDistribution: propTypes.object.isRequired,
}

export default ObservingStatisticsSeeing
