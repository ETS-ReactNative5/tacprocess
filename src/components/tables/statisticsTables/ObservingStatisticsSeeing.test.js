import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ObservingStatisticsSeeing from "./ObservingStatisticsSeeing";

const proposals = [
  {
    transparency: 'Clear',
    maxSeeing: 2,
    instruments: {
      hrs: [ { exposureMode: "HIGH RESOLUTION" } ],
      rss: [ { dictatorMode: 'NORMAL', mode: 'Spectroscopy' } ],
      scam: [ { dictatorMode: "NORMAL" } ],
      bvit: [{type: "BVIT"}]
    },
    requestedTime: {
      minimum: 26350,
      semester: "2018-1",
      requests: { OTH: 0, POL: 85000}
    },
    timeRequests: [
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 150 }, { partnerCode: 'UW', time: 200 }, { partnerCode: 'IUCAA', time: 250 } ]
      },
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 400 }, { partnerCode: 'UW', time: 50 } ]
      }
    ]
  },
  {
    transparency: 'Thin cloud',
    maxSeeing: 2,
    instruments: {
      hrs: [ { exposureMode: "HIGH RESOLUTION" } ],
      rss: [ { dictatorMode: 'NORMAL', mode: 'Spectroscopy' } ],
      scam: [ { dictatorMode: "NORMAL" } ],
      bvit: [{type: "BVIT"}]
    },
    requestedTime: {
      minimum: 26350,
      semester: "2017-2",
      requests: { OTH: 0, POL: 85000}
    },
    timeRequests: [
      {
        semester: '2017-1',
        distribution: [ { partnerCode: 'RSA', time: 150 }, { partnerCode: 'UW', time: 200 }, { partnerCode: 'IUCAA', time: 250 } ]
      },
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 400 }, { partnerCode: 'UW', time: 50 } ]
      }
    ]
  }
];

const partners = ['UW', 'RSA'];

// Checking if the ObservingStatisticsSeeing Component renders correctly for different input
describe("ObservingStatisticsSeeing Component", () => {

  //Enzyme method testing if it renders correctly with empty values of the proposals, semester, and partner
  it("Render the ObservingStatisticsSeeing Component having unpopulated props with no errors", () => {
    const rendered = mount(<ObservingStatisticsSeeing proposals={[]} partner={""} />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });

  // it("Render the ObservingStatisticsSeeing Component having populated props with no errors", () => {
  //   const rendered = mount(<ObservingStatisticsSeeing proposals={proposals} partner={partners[0]} />);
  //   expect(shallowToJson(rendered)).toMatchSnapshot();
  // });
});
