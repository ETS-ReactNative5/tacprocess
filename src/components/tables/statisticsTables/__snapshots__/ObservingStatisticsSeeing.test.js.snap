// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservingStatisticsSeeing Component Render the ObservingStatisticsSeeing Component having unpopulated props with no errors 1`] = `
<ObservingStatisticsSeeing
  partner=""
  proposals={Array []}
>
  <div
    className="stat-item"
    style={
      Object {
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <table
      className="stat-table"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <thead>
        <tr>
          <th
            className="width-150"
          >
            Conditions
          </th>
          <th>
            Time requested (hrs)
          </th>
          <th>
            Number of Proposals
          </th>
          <th>
            Fraction of Total Request (%)
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Max Seeing 
            <br />
             ≦ 1.0 
          </td>
          <td>
            0.00
          </td>
          <td>
            0
          </td>
          <td>
            NaN
          </td>
        </tr>
        <tr>
          <td>
            Max Seeing 
            <br />
             ≦ 1.5
          </td>
          <td>
            0.00
          </td>
          <td>
            0
          </td>
          <td>
            NaN
          </td>
        </tr>
        <tr>
          <td>
            Max Seeing 
            <br />
             ≦ 2.0
          </td>
          <td>
            0.00
          </td>
          <td>
            0
          </td>
          <td>
            NaN
          </td>
        </tr>
        <tr>
          <td>
            Max Seeing 
            <br />
             ≦ 5.0
          </td>
          <td>
            0.00
          </td>
          <td>
            0
          </td>
          <td>
            NaN
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</ObservingStatisticsSeeing>
`;
